server/src/pages/shotindex/controller.js

```diff
@@ -49,16 +52,75 @@ function render() {
   page.render(model);
 }
 
+function extractQueryParamValues(searchKeys) {
+  const o = {};
+  let qs = queryString.parse(window.location.search)
+
+  Object.keys(searchKeys).forEach(x => {
+    if (typeof qs[x] !== "undefined") {
+      o[searchKeys[x]] = qs[x];
+    }
+  });
+
+  return o;
+}
+
+function buildQueryStringFromModel(searchKeys, model) {
+  const queryParams = {};
+
+  Object.keys(searchKeys).forEach(x => {
+    if (typeof model[searchKeys[x]] !== "undefined" && model[searchKeys[x]] !== null) {
+      queryParams[x] = model[searchKeys[x]];
+    }
+  })
+
+  return queryString.stringify(queryParams);
+}
+
 exports.onChangeSearch = function(query) {
   model.defaultSearch = query;
-  let url = `/shots?q=${encodeURIComponent(query)}`;
-  if (!query) {
-    url = "/shots";
-  }
-  window.history.pushState(null, "", url);
+  model.pageNumber = 1;
+  updateHistory({q: query, p: null});
   refreshModel();
 };
 
+exports.onChangePage = function(pageNumber) {
+  pageNumber = (pageNumber && parseInt(pageNumber)) || 1;
```

Comment:
Good catch.  I think it's the remnants of earlier code that I didn't update after some refactoring.  At one point the page number could also come from the query string, making it a string instead of a number.

👍